pkg/manifests/manifests_test.go

```diff
@@ -1188,6 +1164,108 @@ func TestPrometheusK8sRemoteWriteURLs(t *testing.T) {
+	}{
+		{
+			name: "basic authentication configuration",
+			config: `prometheusK8s:
+  remoteWrite:
+  - url: "https://basicAuth.remotewrite.com/api/write"
+    basicAuth:
+      username:
+        name: remoteWriteAuth
+        key: user
+      password:
+        name: remoteWriteAuth
+        key: password
+`,
+			checkFn: []func(*testing.T, monv1.RemoteWriteSpec){
+				func(t *testing.T, target monv1.RemoteWriteSpec) {
+					if target.BasicAuth.Username.Name != "remoteWriteAuth" {
+						t.Fatalf("Name field not correct in section RemoteWriteSpec.BasicAuth.Username expected 'remoteWriteAuth', got %s", target.BasicAuth.Username.Name)
+					}
+					if target.BasicAuth.Username.Key != "user" {
+						t.Fatalf("Key field not correct in section RemoteWriteSpec.BasicAuth.Username expected 'user', got %s", target.BasicAuth.Username.Key)
+					}
+					if target.BasicAuth.Password.Name != "remoteWriteAuth" {
+						t.Fatalf("Name field not correct in section RemoteWriteSpec.BasicAuth.Password expected 'remoteWriteAuth', got %s", target.BasicAuth.Password.Name)
+					}
+					if target.BasicAuth.Password.Key != "password" {
+						t.Fatalf("Key field not correct in section RemoteWriteSpec.BasicAuth.Password expected 'password', got %s", target.BasicAuth.Password.Key)
+					}
+				},
+			},
+		},
+		{
+			name: "bearerTokenFile authentication configuration",
+			config: `prometheusK8s:
+  remoteWrite:
+  - url: "https://bearerTokenFile.remotewrite.com/api/write"
+    bearerTokenFile: "/secret/remoteWriteAuth"
+`,
+			checkFn: []func(*testing.T, monv1.RemoteWriteSpec){
+				func(t *testing.T, target monv1.RemoteWriteSpec) {
+					if target.BearerTokenFile != "/secret/remoteWriteAuth" {
+						t.Fatalf("BearerTokenFile field not correct in section RemoteWriteSpec expected '/secret/remoteWriteAuth', got %s", target.BearerTokenFile)
+					}
+				},
+			},
+		},
+		{
+			name: "authorization authentication configuration",
+			config: `prometheusK8s:
+  remoteWrite:
+  - url: "https://authorization.remotewrite.com/api/write"
+    authorization:
+      type: Bearer
+      credentials:
+        name: remoteWriteAuth
+        key: token
+`,
+			checkFn: []func(*testing.T, monv1.RemoteWriteSpec){
+				func(t *testing.T, target monv1.RemoteWriteSpec) {
+					if target.Authorization.Type != "Bearer" {
+						t.Fatalf("Bearer field not correct in section RemoteWriteSpec expected 'Bearer', got %s", target.Authorization.Type)
+					}
+					if target.Authorization.Credentials.Name != "remoteWriteAuth" {
+						t.Fatalf("Name field not correct in section RemoteWriteSpec.Authorization.Credentials expected 'remoteWriteAuth', got %s", target.Authorization.Credentials.Name)
+					}
+					if target.Authorization.Credentials.Key != "token" {
+						t.Fatalf("Key field not correct in section RemoteWriteSpec.Authorization.Credentials expected 'token', got %s", target.Authorization.Credentials.Key)
+					}
+				},
+			},
+		},
+	} {
+		c, err := NewConfigFromString(tc.config)
```

Comment:
(nit) you can wrap everything withing `t.Run()`

```suggestion
		t.Run(tc.name, func(t *testing.T) {
			c, err := NewConfigFromString(tc.config)
```